galsim/compound.py

```diff
@@ -322,18 +322,26 @@ def __init__(self, *args, **kwargs):
             else:
                 real_space = False
 
-        # Warn if doing DFT convolution for objects with hard edges.
+        # Warn if doing DFT convolution for objects with hard edges
         if not real_space and hard_edge:
-            import warnings
-            if len(args) == 2:
-                msg = """
-                Doing convolution of 2 objects, both with hard edges.
-                This might be more accurate and/or faster using real_space=True"""
-            else:
-                msg = """
-                Doing convolution where all objects have hard edges.
-                There might be some inaccuracies due to ringing in k-space."""
-            warnings.warn(msg)
+            
+            # If any aren't analytic, we don't have a choice here, so suppress the warning
```

Comment:
I think you'll find that your proposed scenario would not emit a warning.  Only if all three have hard edges would there be a warning, and that would be one that lets the user know about possible ringing, not telling them they should use real space (since you can't with 3).

My point is really that the hard_edges attribute should imply analytic_x, in which case the current logic in this function is (I believe) correct.